Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Instance/InstanceDomGenerator.cpp

```diff
@@ -126,6 +135,77 @@ namespace AzToolsFramework
             return true;
         }
 
+        bool InstanceDomGenerator::GenerateEntityDom(PrefabDom& entityDom, const AZ::Entity& entity) const
+        {
+            // Retrieves the focused instance.
+            auto prefabFocusInterface = AZ::Interface<PrefabFocusInterface>::Get();
+            if (!prefabFocusInterface)
+            {
+                AZ_Assert(false, "Prefab - InstanceDomGenerator::GenerateEntityDom - "
+                    "PrefabFocusInterface could not be found.");
+                return false;
```

Comment:
The fact that we are relying on both the output dom and the return value to represent whether the function generated a valid dom seems error prone to me as the bool and the dom state need to be in sync. Instead of returning bool, it can be a void function and the validity can be checked on the dom itself using something like entityDom.IsNull() ?

Reply:
+1 Agree. Along with the comment on DOM check above, I will put up a follow-up PR for these changes as it also involves changes in InstanceDomGenerator, which is unrelated to this PR.